### PR TITLE
[Docs] Add search timeout caveats

### LINKED ISSUE #31263: [DOCS] Clarify that search timeouts behave like cancellation
The documentation for the [global search timeout](https://www.elastic.co/guide/en/elasticsearch/reference/current/search.html#global-search-timeout) setting does not describe how strictly the timeout applies. In fact, it seems that the timeout works on a polling basis, similarly to [search cancellation](https://www.elastic.co/guide/en/elasticsearch/reference/current/search.html#global-search-cancellation), and therefore a search may take substantially longer than the configured timeout, and it may even do so successfully rather than timing out.

I think it'd be helpful to add this information to this documentation to clarify the situation.

### PR DESCRIPTION
**Pull request by cbuescher**
Global search timeouts and timeouts specified in the search request body use the
same internal mechanism as search cancellation. Therefore the same caveats
apply, mostly around the responsiveness of the timeout which gets only checked
by a running search on segment boundaries by default.

Closes #31263

### CODE REVIEW

**Comment by cbuescher** [comment]
@DaveCTurner do you think this addition adresses the issue you opened, or do you think it needs to be more specific?

File: docs/reference/search.asciidoc [hunk 1]
```diff
@@ -119,10 +119,14 @@ Individual searches can have a timeout as part of the
 <<search-request-body>>. Since search requests can originate from many
 sources, Elasticsearch has a dynamic cluster-level setting for a global
 search timeout that applies to all search requests that do not set a
-timeout in the <<search-request-body>>. The default value is no global
-timeout. The setting key is `search.default_search_timeout` and can be
-set using the <<cluster-update-settings>> endpoints. Setting this value
-to `-1` resets the global search timeout to no timeout.
+timeout in the request body. These requests will be canceled after
```
Comment (colings86) [inline docs/reference/search.asciidoc]: `canceled` -> `cancelled`

**Review by colings86 — APPROVED** [review]
LGTM - left one typo comment

**Comment by DaveCTurner** [comment]
Thanks @cbuescher, this looked good to me too (with both US and UK spellings of cancelled).

**Pull request merged**